File: app/src/main/java/com/example/myapplication/noteapp/model/repository/NoteRepository.java
package com.example.myapplication.noteapp.model.repository;

import android.content.Context;

import androidx.lifecycle.LiveData;

import com.example.myapplication.noteapp.model.database.AppDao;
import com.example.myapplication.noteapp.model.database.NoteRoomDataBase;
import com.example.myapplication.noteapp.model.database.entity.Category;
import com.example.myapplication.noteapp.model.database.entity.CategoryWithNotes;
import com.example.myapplication.noteapp.model.database.entity.Note;

import java.util.List;
import java.util.concurrent.ExecutorService;

import javax.inject.Inject;

public class NoteRepository {
    private AppDao appDao;
    private ExecutorService service;

    @Inject
    public NoteRepository(AppDao appDao, ExecutorService service){
        this.appDao = appDao;
        this.service = service;
    }

    public void insertCategory(Category category){service.execute(() -> appDao.insertCategory(category));}

    public void insertNote(Note note){
        service.execute(() -> appDao.insertNote(note));
    }

    public void deleteNote(Note note){
        service.execute(() -> appDao.deleteNote(note));
    }

    public void updateNote(Note note){
        service.execute(() -> appDao.updateNote(note));
    }

    public LiveData<CategoryWithNotes> getNotesByCategory(String nameCategory){return appDao.getNotesByCategory(nameCategory);}

    public LiveData<List<Note>> getNotesBySubtext(String searchText){return appDao.getNotesBySubtext(searchText);}

    public LiveData<List<Note>> getAllNotes(){
        return appDao.getAllNotes();
    }

    public LiveData<List<Category>> getAllCategories(){
        return appDao.getCategories();
    }
}

File: lib/data/repository/note_repository.dart
import 'package:get_it/get_it.dart';
import 'package:note_app/data/local/note.dart';
import 'package:note_app/data/local/note_database.dart';

class NoteRepository {
  final NoteDatabase _database = GetIt.instance<NoteDatabase>();

  Future<void> insertNote(Note note) async {
    await _database.noteDao.insertNote(note);
  }

  Future<void> deleteNote(Note note) async {
    await _database.noteDao.deleteNote(note);
  }

  Future<void> editNote(Note note) async {
    await _database.noteDao.updateNote(note);
  }

  Future<List<Note>> getAllNotes() async {
    return await _database.noteDao.getNotes() ?? [];
  }

  Future<List<Note>> searchNoteBySearchText(String searchText) async {
    return await _database.noteDao.searchNotes(searchText) ?? [];
  }
}
